primap2/pm2io/_data_reading.py

```diff
@@ -539,6 +711,64 @@ def filter_data(
     data.reset_index(drop=True, inplace=True)
 
 
+def fill_from_other_col(
+    df: pd.DataFrame,
+    *,
+    coords_value_filling: Dict[str, Dict[str, Dict[str, str]]],
+    attrs: Dict[str, Any],
+) -> pd.DataFrame:
+    """
+    This function fills value in one column based on values in other columns.
+    It can be used to fill NaN values or to replace e.g. non-standard or
+    non-unique category codes based on category names. It operates on pandas
+    DataFrames.
+
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Data to operate on
+
+    coords_value_filling : dict
+        A dict with primap2 dimension names as keys. These are the target columns where
+        values will be filled (or replaced). Vales are dicts with primap2 dimension
+        names as keys. These are the source columns. The values are dicts with source
+        value - target value mappings.
+        This can be used to e.g. fill missing category codes based on category names or
+        to replace category codes which do not meet the terminology using the category
+        names.
+
+    attrs : dict
+        Dataset attributes
+
+    Returns
+    -------
+    pd.DataFrame
+    """
+    df_filled = df.copy()
```

Comment:
The copying is not needed, I think. We don't copy in any of the mapping functions, either, so I think we can just operate in place here as well.

Reply:
OK, will change that. It's probably a leftover from testing in a notebook

Reply:
done